src/test/groovy/org/sonarqube/gradle/GradleKtsTests.groovy

```diff
@@ -0,0 +1,162 @@
+        Files.copy(is, testProjectDir.resolve('gradle.properties'), StandardCopyOption.REPLACE_EXISTING)
+    }
+
+    def "add build and settings gradle files to sources when both are in kotlin dsl"() {
+        given:
+        addBuildKts()
+        addSettingsKts()
+
+        when:
+        def result = GradleRunner.create()
+                .withProjectDir(testProjectDir.toFile())
+                .forwardOutput()
+                .withArguments('sonarqube', '--info', '-Dsonar.scanner.dumpToFile=' + outFile.toAbsolutePath())
+                .withPluginClasspath()
+                .build()
+
+        def props = new Properties()
+        props.load(outFile.newDataInputStream())
+
+        then:
+        def sonarSources = props["sonar.sources"].split(",")
+        Assertions.assertThat(sonarSources)
+                .containsExactlyInAnyOrder(settingsFile.toRealPath().toString(), buildFile.toRealPath().toString())
+
+    }
+
+    def "add only build and file to sources when only build is in kotlin dsl"() {
+        given:
+        addBuildKts()
+        addSettings()
+
+        when:
+        def result = GradleRunner.create()
+                .withProjectDir(testProjectDir.toFile())
+                .forwardOutput()
+                .withArguments('sonarqube', '--info', '-Dsonar.scanner.dumpToFile=' + outFile.toAbsolutePath())
+                .withPluginClasspath()
+                .build()
+
+        def props = new Properties()
+        props.load(outFile.newDataInputStream())
+
+        then:
+        props["sonar.sources"] == buildFile.toRealPath().toString()
+
+    }
+
+    def "add only settings file to sources when only settings file is in kotlin dsl"() {
+        given:
+        addBuild()
+        addSettingsKts()
+
+        when:
+        def result = GradleRunner.create()
+                .withProjectDir(testProjectDir.toFile())
+                .forwardOutput()
+                .withArguments('sonarqube', '--info', '-Dsonar.scanner.dumpToFile=' + outFile.toAbsolutePath())
+                .withPluginClasspath()
+                .build()
+
+        def props = new Properties()
+        props.load(outFile.newDataInputStream())
+
+        then:
+        props["sonar.sources"] == settingsFile.toRealPath().toString()
+
+    }
+
+    def "add nothing to sources when Groovy dsl is used"() {
+        given:
+        addBuild()
+        addSettings()
+
+        when:
+        def result = GradleRunner.create()
+                .withProjectDir(testProjectDir.toFile())
+                .forwardOutput()
+                .withArguments('sonarqube', '--info', '-Dsonar.scanner.dumpToFile=' + outFile.toAbsolutePath())
+                .withPluginClasspath()
+                .build()
+
+        def props = new Properties()
+        props.load(outFile.newDataInputStream())
+
+        then:
+        props["sonar.sources"] == ""
+
+    }
+
```

Comment:
Maybe add two more test cases:
1. with buildKts, but without settings (neither KTS nor Grooy)
2. without both build and settings

Reply:
About 1, I can add tests,
About 2 - we can't execute Gradle scanner if it's not a Gradle project, so there should be always build file